samples: bluetooth: fix no ble advertising after disconnecting

StefJar opened the pull request:
Related to #90111

At the ble GATT samples the advertising is not restarted after a disconnect happen. That fix adds that feature at the right place(at the recycle callback).

Commit:
samples: bluetooth: fix no advertising after disconnecting

Related to #90111

At the ble GATT samples the advertising is not restarted after
a disconnecting happend. That fix addes that feature at the
right place(at the recycle callback).

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

Commit:
samples: bluetooth: periphal(_hr) fix clang format and compiler error

fix:
	- clang format errors
	- little bug that compares a return value instead of assigning it

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

Commit:
samples: bluetooth: fix missing variable declaration

fix:
	- missing err variable declaration

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

Commit:
samples: bluetooth: fix periphal hr restart advertising

fix:
	- at ext. adv we don't need to rebuild the adv payload

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

StefJar commented:
sorry - its abit messy since I don't have a board for testing here. So I am using you test bed to check the code. I will get there. ;)

alwa-nordic commented:
> sorry - its abit messy since I don't have a board for testing here. So I am using you test bed to check the code. I will get there. ;)

FYI there are no hardware tests in Zephyr CI. Everything is simulated. You can run the tests locally without any boards.

Commit:
samples: bluetooth: fix advertising type for ext adv

fix:
	- different adv variables for normal and ext. adv

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

Commit:
samples: bluetooth: periphal_hr fix broken EXT_AVT sample

fix:
	- missing else branch

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

StefJar commented:
@alwa-nordic looks like I am finally through with the code
now the compliance test fails
I used the current zephyr clang formater file
maybe you can have a look and advice me

Thalley commented on samples/bluetooth/peripheral/src/main.c:
```diff
@@ -236,9 +214,7 @@ void mtu_updated(struct bt_conn *conn, uint16_t tx, uint16_t rx)
-static struct bt_gatt_cb gatt_callbacks = {
-	.att_mtu_updated = mtu_updated
-};
+static struct bt_gatt_cb gatt_callbacks = {.att_mtu_updated = mtu_updated};
```
```suggestion
static struct bt_gatt_cb gatt_callbacks = {.att_mtu_updated = mtu_updated, };
```

Thalley commented on samples/bluetooth/peripheral/src/main.c:
```diff
@@ -254,6 +230,16 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 
+static void recycled(void)
+{
+	printk("connection recycled. Restart advertising a connection");
```
```suggestion
	printk("Connection recycled. Restarting advertisements\n");
```

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -61,17 +73,26 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	(void)atomic_set_bit(state, STATE_DISCONNECTED);
 }
 
+static void recycled(void)
```
```suggestion
static void recycled_cb(void)
```

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -61,17 +73,26 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 
+static void recycled(void)
+{
+	printk("connection recycled. Restart advertising a connection");
```
```suggestion
	printk("Connection recycled. Restarting advertisements\n");
```

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -61,17 +73,26 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
+static void recycled(void)
+{
+	printk("connection recycled. Restart advertising a connection");
+	const int err = start_adv();
+	if (err) {
```
```suggestion
	const int err = start_adv();
	
	if (err) {
```

Thalley replied:
It's required by our coding guidelines :)

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -61,17 +73,26 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
+static void recycled(void)
+{
+	printk("connection recycled. Restart advertising a connection");
+	const int err = start_adv();
+	if (err) {
```
```suggestion
	const int err = start_adv();
	if (err != 0) {
```

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -186,6 +207,28 @@ static void blink_stop(void)
+	int err;
+#if !defined(CONFIG_BT_EXT_ADV)
+	printk("Starting Legacy Advertising (connectable and scannable)\n");
+	err = bt_le_adv_start(BT_LE_ADV_CONN_FAST_1, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (err) {
```
```suggestion
	err = bt_le_adv_start(BT_LE_ADV_CONN_FAST_1, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
	if (err != 0) {
```

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -186,6 +207,28 @@ static void blink_stop(void)
 
+static int start_adv(void)
+{
+	int err;
+#if !defined(CONFIG_BT_EXT_ADV)
```
```suggestion
	int err;
	
#if !defined(CONFIG_BT_EXT_ADV)
```

Thalley replied:
Our coding guidelines require an empty line after declarations

Thalley commented on samples/bluetooth/peripheral_hr/src/main.c:
```diff
@@ -186,6 +207,28 @@ static void blink_stop(void)
+#else  /* CONFIG_BT_EXT_ADV */
+	printk("Starting Extended Advertising (connectable and non-scannable)\n");
+	err = bt_le_ext_adv_start(adv, BT_LE_EXT_ADV_START_DEFAULT);
+	if (err) {
```
```suggestion
	if (err != 0) {
```

Thalley reviewed (changes requested):
Personally I'd keep this as 2 commits:
1 commit to update the formatting (instead of the current 5 commits that does this...)
1 commit to fix the advertising issue that you are trying to fix. 

Alternatively, and perhaps better, just keep the commit that fixes the issue, instead of formatting the entire files.

Commit:
samples: bluetooth: added post fix for callbacks

changes:
	- added postfix "_cb" for ble related callback functions
	- changed err code checking form (err) to (0 != err)

Signed-off-by: Stefan Jaritz <stefan.jaritz@gmail.com>

Thalley commented on samples/bluetooth/peripheral/src/main.c:
```diff
@@ -95,17 +91,14 @@ static void indicate_destroy(struct bt_gatt_indicate_params *params)
+	'd', 'o', 'r', ' ', 'd', 'a', 't', 'a', '2', 'V', 'e', 'n', 'd', 'o', 'r',
+	' ', 'd', 'a', 't', 'a', '3', 'V', 'e', 'n', 'd', 'o', 'r', ' ', 'd', 'a',
+	't', 'a', '4', 'V', 'e', 'n', 'd', 'o', 'r', ' ', 'd', 'a', 't', 'a', '5',
+	'V', 'e', 'n', 'd', 'o', 'r', ' ', 'd', 'a', 't', 'a', '6', '.', ' '};
```
```suggestion
	'V', 'e', 'n', 'd', 'o', 'r', ' ', 'd', 'a', 't', 'a', '6', '.', ' ', };
```

It helps with clang-format if you always have a final `,`

Thalley commented on samples/bluetooth/peripheral/src/main.c:
```diff
@@ -187,73 +176,70 @@ static ssize_t write_without_rsp_vnd(struct bt_conn *conn,
+static void recycled_cb(void)
+{
+	printk("connection recycled. Restart advertising a connection");
+	const int err =
+		bt_le_adv_start(BT_LE_ADV_CONN_FAST_1, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (0 != err) {
```
```suggestion
	const int err =
		bt_le_adv_start(BT_LE_ADV_CONN_FAST_1, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
		
	if (err != 0) {
```

Thalley reviewed (changes requested):
Suggest to squash all formatting commit to a single commit

StefJar commented:
Thx for @Thalley his effort and sorry to create workload
I used the clang-format file from the repro. Looks like it is not fully working. Is there any script that does the code formatting? BR
